micromanager_gui/main_window.py

```diff
@@ -176,17 +306,32 @@ def _on_mda_frame(self, image: np.ndarray, event: useq.MDAEvent):
             for a, v in enumerate(im_idx):
                 self.viewer.dims.set_point(a, v)
 
+            #save each image in the temp folder, 
+            image_name = f'{im_idx}.tif'
+            savefile = Path(self.temp_folder.name) / image_name
```

Comment:
test is [failing](https://github.com/tlambert03/napari-micromanager/pull/13/checks?check_run_id=2786248572#step:8:132) with an attribute error because `self.temp_folder` will _only_ exist if another method `_on_mda_started` has been called first ...  (when I wrote the [test](https://github.com/tlambert03/napari-micromanager/blob/c8e1fbba4cc63dee5ac976608e764816fcbc34f9/tests/test_main_window.py#L43), `_on_mda_started` didn't exist)

In general it's a good idea to try to make sure that class methods don't need to be called in "just the right order".  either put a `self.temp_folder = None` thing in the `__init__` function, or check here for `if hasattr(self, 'temp_folder')` ...

Reply:
I added:

```
if hasattr(self, 'temp_folder'):
     savefile = Path(self.temp_folder.name) / image_name
     tifffile.tifffile.imsave(str(savefile), image)
```

but there is still an error. Looking in the error details:

```
 File "/Users/runner/work/napari-micromanager/napari-micromanager/micromanager_gui/main_window.py", line 459, in change_objective
32
    raise ValueError (
33
ValueError: MAGNIFICATION NOT SET, STORE OBJECTIVES NAME STARTING WITH e.g. 100X or 100x.
```

Reply:
that's because you changed it to an error [in this PR](https://github.com/tlambert03/napari-micromanager/pull/13/files#diff-649c4a93f80a0ec181a3ef9ae704d234cd8d7745d59dba35220e94166ed3f428R459) :joy:

I think a value error is a bit strict there.  I don't think we want to make any assumptions about the config file, like "you must store objectives with a specific name".  I'd go back to a print statement and put that whole thing on the todo list for another PR

Reply:
Yes but since the code is doing what it is supposed to do, shouldn't it be considered as a "non error"?

Reply:
maybe it's working for your config but not for the demo config provided by micromanager? :)

Reply:
basically... for some reason or another, it _is_ erroring here.  so the challenge (and the whole purpose/benefit of having tests here ...) is to figure out what assumption or code caused the test to fail, where it was passing before.  Sometimes you'll find that the test itself is in error, other times you'll find that it caught something you hadn't anticipated.
In this case, I suspect that you typically develop with one very specific demo config, and it works for that case... but that's not what is being tested here.

Reply:
> maybe it's working for your config but not for the demo config provided by micromanager? :)

Yes it works only with the demo_config I created just because from the names of the objectives, the code is able to extract the magnification and so calculate the pixel size.

Reply:
right... and I'm saying that's a bit _too_ strict.  In other words, we can't tell someone using this program their config file, which is totally valid and works in micromanger, is no longer valid here... and raise a `ValueError`.  You can certainly _look_ for some sort of convention and print a warning or something if you don't find it (but even there I think we should try to figure out if it can be done without using a special naming convention)... But you can't actually raise an exception and stop everything.

In any case, the point here is that you need to revert line 458 either back to a print statement, or use 
```python
import warnings

warnings.warn('some warning')
```

anytime you use `raise` ... even if it's `raise Warning` ... you're going to completely stop execution of the program.
https://docs.python.org/3/tutorial/errors.html#raising-exceptions

Reply:
Thanks!!!

I was reading about that and I changed to "Warning" to see what was happening...I'll read a bit more about that